File: app/src/main/java/z/nova/rifmobolgame/RulesGame.java
package z.nova.rifmobolgame;

import android.app.Dialog;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class RulesGame extends AppCompatActivity {

    Dialog dialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.rules);

        Window w = getWindow();
        w.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        Button button_back = (Button) findViewById(R.id.button_back);
        button_back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    Intent intent = new Intent(RulesGame.this, GameMenu.class);
                    startActivity(intent); finish();
                }catch (Exception e){
                    //Пусто
                }
            }
        });

    }
    //Системная кнопка "Назад" - начало
    @Override
    public void onBackPressed(){
        try {
            Intent intent = new Intent(RulesGame.this, GameMenu.class); //Создали намерение для перехода
            startActivity(intent); finish();
        }catch (Exception e){

        }
    }
    //Системная кнопка "Назад" - конец
}

File: app/src/main/java/z/nova/rifmobolgame/RulesGame.kt
package z.nova.rifmobolgame

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class RulesGame : AppCompatActivity() {

    private lateinit var btnBack: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.rules)

        btnBack = findViewById(R.id.button_back)
        btnBack.setOnClickListener {
                val intent = Intent(this@RulesGame, GameMenu::class.java)
                startActivity(intent)
                finish()
        }
    }

    //Системная кнопка "Назад" - начало
    override fun onBackPressed() {
            val intent =
                Intent(this@RulesGame, GameMenu::class.java) //Создали намерение для перехода
            startActivity(intent)
            finish()
    } //Системная кнопка "Назад" - конец
}